dali/kernels/CMakeLists.txt

```diff
@@ -15,6 +15,7 @@
 project(dali_kernels CUDA CXX C)
 
 set(CMAKE_CUDA_SEPARABLE_COMPILATION ON)
+set(CMAKE_CUDA_RESOLVE_DEVICE_SYMBOLS ON)
```

Comment:
What is that for?

Reply:
https://cmake.org/cmake/help/latest/prop_tgt/CUDA_RESOLVE_DEVICE_SYMBOLS.html#prop_tgt:CUDA_RESOLVE_DEVICE_SYMBOLS
This forces the library to link using nvcc in newer CMakes which is able to resolve the symbols (as clang apparently is not able to perform device linking).  Still I use the dummy.cu in older CMakes, so probably this line can be removed, I will check it.